cli/internal/cmd/upgradeexecute.go

```diff
@@ -59,27 +62,35 @@ func runUpgradeExecute(cmd *cobra.Command, args []string) error {
 }
 
 func upgradeExecute(cmd *cobra.Command, imageFetcher imageFetcher, upgrader cloudUpgrader, fileHandler file.Handler) error {
-	configPath, err := cmd.Flags().GetString("config")
+	flags, err := parseUpgradeExecuteFlags(cmd)
 	if err != nil {
-		return err
+		return fmt.Errorf("parsing flags: %w", err)
 	}
-	conf, err := config.New(fileHandler, configPath)
+	conf, err := config.New(fileHandler, flags.configPath)
 	if err != nil {
 		return displayConfigValidationErrors(cmd.ErrOrStderr(), err)
 	}
 
-	helm, err := cmd.Flags().GetBool("helm")
-	if err != nil {
-		return err
-	}
-	if helm {
-		timeout, err := cmd.Flags().GetDuration("timeout")
-		if err != nil {
-			return err
+	if flags.helmFlag {
+		if !flags.yes {
+			err = upgrader.UpgradeHelmServices(cmd.Context(), conf, flags.upgradeTimeout, helm.DenyDestructive)
+			if errors.Is(err, helm.ErrConfirmationMissing) {
+				cmd.PrintErrln("WARNING: Upgrading cert-manager will destroy all custom resources you have manually created that are based on the current version of cert-manager.")
+				ok, askErr := askToConfirm(cmd, "Do you want to upgrade cert-manager anyway?")
+				if askErr != nil {
+					return fmt.Errorf("asking for confirmation: %w", err)
+				}
+				if !ok {
+					cmd.Println("Aborting upgrade.")
+					return nil
+				}
+			}
```

Comment:
As far as I can tell the UpgradeHelmServices function is currently called twice in case the yes flag isn't set.